File: src/main/java/com/timxyz/services/LocationTypeService.java
package com.timxyz.services;

import com.timxyz.models.Location;
import com.timxyz.models.LocationType;
import com.timxyz.repositories.LocationTypeRepository;
import com.timxyz.services.exceptions.ServiceException;

import java.util.Collection;

import org.springframework.stereotype.Service;

@Service
public class LocationTypeService extends BaseService<LocationType, LocationTypeRepository> {
    public LocationType save(LocationType model) throws ServiceException {
        LocationType sameName = getByName(model.getName());

        if (sameName != null && model.getId() != sameName.getId()) {
            throw new ServiceException("Tip lokacije s navedenim imenom već postoji.");
        }

        return super.save(model);
    }

    public void delete(Long id) throws ServiceException {
        LocationType lt = get(id);

        if (lt.getLocations().size() > 0) {
            throw new ServiceException("Tip lokacije se ne može obrisati jer je vezan za neke od postojećih lokacija.");
        }

        super.delete(id);
    }

    public Collection<LocationType> filterByName(String name) {
        return repository.filterByName(name);
    }
    
    public LocationType getByName(String name) {
        return repository.findFirstByName(name);
    }
}

File: frontend/client/app/services/locationTypeService.js
import BaseService from './baseService';

export default class LocationTypeService extends BaseService {
	all() {
		return super.get('locationTypes/all');
	}

	create(locType) {
		return super.post('locationTypes', locType);
	}

	getPage(page){
		return super.get('locationTypes/page/'+page);
	}

	delete(id) {
		return super.delete('locationTypes/' + id);
	}

	find(id) {
        return super.get('locationTypes/' + id);
    }

    update(id, locationType) {
        return super.post('locationTypes/' + id, locationType);
    }

    filterByName(name) {
        return super.get('locationTypes/filter-by/name?name=' + name);
    }
}
